src/peft/tuners/cpt/config.py

```diff
@@ -79,7 +80,12 @@ def __post_init__(self):
         self.num_attention_heads = None  # Number of attention heads (if applicable).
         self.num_transformer_submodules = 1  # Number of transformer submodules used.
         self.peft_type = PeftType.CPT  # Specifies that the PEFT type is CPT.
-        self.task_type = "CAUSAL_LM"  # Ensures task type is causal language modeling.
+        if self.task_type != TaskType.CAUSAL_LM:
+            warnings.warn(
+                f"{self.__class__.__name__} only supports task_type = {TaskType.CAUSAL_LM.value}, "
+                "setting it automatically."
+            )
+            self.task_type = TaskType.CAUSAL_LM  # Ensures task type is causal language modeling.
```

Comment:
Is erroring out not a better option here?

Reply:
Okay. If that is the case, I think it might make sense to start a deprecation cycle (can be done in a future PR).

Reply:
This is a good idea, I changed the code to give a `FutureWarning`